src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/BlockingCollection.cs

```diff
@@ -755,7 +755,16 @@ private bool TryTakeWithNoTimeValidation([MaybeNullWhen(false)] out T item, int
                     }
                 }
             }
+
+            if (waitForSemaphoreWasSuccessful)
+            {
+                Debug.Assert(item != null);
+            }
+
+#pragma warning disable CS8762
+            // Compiler can't automatically deduce 'item' has a non-null value when returning false.
```

Comment:
I don't understand the comment.  `item` is allowed to be null if the method returns false.